src/Sarif.Multitool/Rules/SARIF1019.RuleIdMustBePresentAndConsistent.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class RuleIdMustBePresentAndConsistent : SarifValidationSkimmerBase
+    {
+        private readonly MultiformatMessageString _fullDescription = new MultiformatMessageString
+        {
+            Text = RuleResources.SARIF1019_MissingResultRuleId
```

Comment:
>SARIF1019_MissingResultRuleId [](start = 33, length = 29)

The "full description" string that describes the _rule_ is distinct from any of the message strings that describe individual _results_ from the rule. There needs to be a separate resource string, whose name matches the rule (`SARIF1019_RuleIdMustBePresentAndConsistent`). Let's give it this text:

In every result, at least one of the properties result.ruleId and result.rule.id must be present. If both are present, they must be equal.

Reply:
ok!